### PR TITLE
fix: fix crash when `message.fix` is nullish

### PR DESCRIPTION
**Pull request by ntnyq**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

No need for providing the template. So I skipped it.

#### What changes did you make? (Give an overview)

This PR adds a simple check in `linter.source-code-fixer` incasing `message.fix` equals to `null` or `undefined` which might cause the linter throw. 

#### Is there anything you'd like reviewers to focus on?

There is a reproduction repo https://github.com/pany-ang/issue-eslint-config.

Or you can just remove the check, run the tests added.

### CODE REVIEW

**Comment by nzakas** [comment]
Thanks for the PR. It's not clear from the changes or from you repo how to reproduce this error. Can you please provide some reproduction steps?

**Comment by nzakas** [comment]
@ntnyq friendly ping. Can you answer my questions above? Thanks.

**Comment by ntnyq** [comment]
> @ntnyq friendly ping. Can you answer my questions above? Thanks.

Sorry for reply late.

I created a new repo with fewer code https://github.com/ntnyq-dev/repro-eslint-fix.

And added the repro steps in README.

**Comment by ntnyq** [comment]
As I tested.

When call `SourceCodeFixer.applyFixes`, there is a message like this:

```
{
      ruleId: 'format/prettier',
      severity: 2,
      message: 'Parsing error: CssSyntaxError: Unclosed block',
      line: 6,
      column: 3,
      nodeType: null,
      endLine: 1,
      endColumn: NaN,
      fix: undefined
}
```

It cause the crash.

**Comment by nzakas** [comment]
Thanks, I can reproduce the crash. For the record, here's what I see in the console:

```
$ npm run lint

> repro-eslint-fix@1.0.0 lint
> eslint . --fix


Oops! Something went wrong! :(

ESLint: 9.16.0

TypeError: Cannot read properties of undefined (reading 'range')
    at attemptFix (C:\Users\nzaka\projects\eslint\tmp\repro-eslint-fix\node_modules\eslint\lib\linter\source-code-fixer.js:88:27)
    at SourceCodeFixer.applyFixes (C:\Users\nzaka\projects\eslint\tmp\repro-eslint-fix\node_modules\eslint\lib\linter\source-code-fixer.js:123:17)
    at Linter.verifyAndFix (C:\Users\nzaka\projects\eslint\tmp\repro-eslint-fix\node_modules\eslint\lib\linter\linter.js:2329:43)
    at verifyText (C:\Users\nzaka\projects\eslint\tmp\repro-eslint-fix\node_modules\eslint\lib\eslint\eslint.js:329:48)
    at C:\Users\nzaka\projects\eslint\tmp\repro-eslint-fix\node_modules\eslint\lib\eslint\eslint.js:810:40
```

In the future, please be sure to open an issue first when reporting a bug so you can provide all of the info we need.

File: tests/lib/linter/source-code-fixer.js [hunk 1]
```diff
@@ -421,6 +429,22 @@ describe("SourceCodeFixer", () => {
 
         });
 
+        describe("Nil fix", () => {
```
Comment (nzakas) [inline tests/lib/linter/source-code-fixer.js:432]:
```suggestion
        describe("Nullish fixes", () => {
```

File: tests/lib/linter/source-code-fixer.js [hunk 2]
```diff
@@ -421,6 +429,22 @@ describe("SourceCodeFixer", () => {
...
+            it("should not throw if fix equal null", () => {
```
Comment (nzakas) [inline tests/lib/linter/source-code-fixer.js:433]:
```suggestion
            it("should not throw if fix is null", () => {
```

File: tests/lib/linter/source-code-fixer.js [hunk 3]
```diff
@@ -421,6 +429,22 @@ describe("SourceCodeFixer", () => {
...
+                const result = SourceCodeFixer.applyFixes(TEST_CODE, [NULL_FIX]);
+
+                assert.isFalse(result.fixed);
+                assert.strictEqual(result.output, TEST_CODE);
+            });
+
+            it("should not throw if fix equal undefined", () => {
```
Comment (nzakas) [inline tests/lib/linter/source-code-fixer.js:440]:
```suggestion
            it("should not throw if fix is undefined", () => {
```

**Review by nzakas — COMMENTED** [review]
Just want to clean up the test names a bit, but otherwise looks good.

**Comment by ntnyq** [comment]
Don't know why Browser Test failed in CI.

I tried run it in local and it went well.

**Comment by nzakas** [comment]
Don't worry about the browser test. It's pretty flaky lately.

**Review by nzakas — DISMISSED** [review]
LGTM. Would like another review before merging.

**Comment by mdjermanovic** [comment]
I think this is rather a bug in the processor.

It gets a lint message without `fix` property:

```
{
    ruleId: 'format/prettier',
    severity: 2,
    message: 'Parsing error: CssSyntaxError: Missed semicolon',
    line: 4,
    column: 12,
    nodeType: null
}
```

but returns message with `fix: undefined`:

```
{
    ruleId: 'format/prettier',
    severity: 2,
    message: 'Parsing error: CssSyntaxError: Missed semicolon',
    line: 17,
    column: 2,
    nodeType: null,
    endLine: 1,
    endColumn: NaN,
    fix: undefined
}
```

Lint messages normally don't have the `fix` property when they're not autofixable.

**Comment by mdjermanovic** [comment]
Now I see that both the [docs](https://eslint.org/docs/latest/integrate/nodejs-api#-lintmessage-type) and [types](https://github.com/eslint/eslint/blob/b607ae64913ca2b6450a74344ab0ad548e314915/lib/types/index.d.ts#L1191) say that the `fix` property can have `undefined` value, so although that can happen only when processors modify messages, seems we should support it.

File: tests/lib/linter/source-code-fixer.js [hunk 4]
```diff
@@ -421,6 +429,22 @@ describe("SourceCodeFixer", () => {
 
         });
 
+        describe("Nullish fixes", () => {
+            it("should not throw if fix is null", () => {
+                const result = SourceCodeFixer.applyFixes(TEST_CODE, [NULL_FIX]);
+
+                assert.isFalse(result.fixed);
+                assert.strictEqual(result.output, TEST_CODE);
```
Comment (mdjermanovic) [inline tests/lib/linter/source-code-fixer.js:437]:
```suggestion
                assert.strictEqual(result.output, TEST_CODE);
                assert.strictEqual(result.messages.length, 1);
                assert.strictEqual(result.messages[0].message, "null");
```

To ensure the message is not removed.

File: tests/lib/linter/source-code-fixer.js [hunk 5]
```diff
@@ -421,6 +429,22 @@ describe("SourceCodeFixer", () => {
...
+            });
+
+            it("should not throw if fix is undefined", () => {
+                const result = SourceCodeFixer.applyFixes(TEST_CODE, [UNDEFINED_FIX]);
+
+                assert.isFalse(result.fixed);
+                assert.strictEqual(result.output, TEST_CODE);
```
Comment (mdjermanovic) [inline tests/lib/linter/source-code-fixer.js:444]:
```suggestion
                assert.strictEqual(result.output, TEST_CODE);
                assert.strictEqual(result.messages.length, 1);
                assert.strictEqual(result.messages[0].message, "undefined");
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**